Array_Vector_Exercises/Exercise1/TDinamicArray.h

```diff
@@ -0,0 +1,242 @@
+	bool IsEmpty() const
+	{
+		return SizeArray == 0;
+	};
+
+	//-------------------------------------------------------------------------
+	int GetSize() const
+	{
+		return SizeArray;
+	}
+
+	//-------------------------------------------------------------------------
+	int GetCapacity() const
+	{
+		return CapacityArray;
+	}
+
+	//-------------------------------------------------------------------------
+	void Reserve(const int NewCapacity)
+	{
+		if (NewCapacity > CapacityArray)
+		{
+			T* ContainerArray = new T[NewCapacity];
+
+			for (int i = 0; i < SizeArray; i++)
+			{
+				ContainerArray[i] = DynamicArray[i];
+			}
+			delete[] DynamicArray;
+
+			DynamicArray = ContainerArray;
+			CapacityArray = NewCapacity;
+		}
+	}
+
+	//--------------------------------------------------------------------------
+	void ShrinkToFit()
+	{
+		if (CapacityArray != SizeArray)
+		{
+			T* ContainerArray = new T[SizeArray];
+
+			for (int i = 0; i < SizeArray; i++)
+			{
+				ContainerArray[i] = DynamicArray[i];
+			}
+			delete[] DynamicArray;
+
+			DynamicArray = ContainerArray;
+		}
+	}
+
+	//--------------------------------------------------------------------------
+	void Clear()
+	{
+		SizeArray = 0;
+	}
+
+	//--------------------------------------------------------------------------
+	void PushBack(const T& InType)
+	{
+		if (SizeArray == CapacityArray)
+		{
+			Reserve(CapacityArray * 2);
+			DynamicArray[SizeArray] = InType;
+			SizeArray++;
+		}
+		else
+		{
+			DynamicArray[SizeArray] = InType;
+			SizeArray++;
+		}
+	}
+
+	//-----------------------------------------------------------------------------
+	void PopBack()
+	{
+		if (SizeArray != 0)
+		{
+			SizeArray--;
+		}
+	}
+
+	//-----------------------------------------------------------------------------
+	void Insert(const int Index, const T& InType)
+	{
+		if (Index < SizeArray && Index >= 0)
+		{
+			if (SizeArray == CapacityArray)
+			{
+				Reserve(CapacityArray * 2);
+			}
+				
+			for (int i = SizeArray; i > Index; i--)
+			{
+				DynamicArray[Index] = DynamicArray[i - 1];
+			}
+			DynamicArray[Index] = InType;
+			SizeArray++;	
+		}
+	}
+
+	//-----------------------------------------------------------------------------
+	void Erase(const int Index)
+	{
+		for (int i = 0; i < SizeArray - 1; i++)
+		{
+			DynamicArray[Index] = DynamicArray[Index + 1];
+		}
+		SizeArray--;
+	}
+
+	//----------------------------------------------------------------------------
+	void Resize(const int NewSize)
+	{
+		SizeArray = NewSize;
+	}
+
+	//--------------------------------------------------------------------------
+	void Append(const TDynamicArray<T>& AppendArray)
+	{
+		Reserve(AppendArray.GetSize() + SizeArray);
+		for (int i = 0; i < AppendArray.GetSize(); i++)
+		{
+			PushBack(AppendArray[i]);
+		}
+	}
+
+	//----------------------------------------------------------------------------
+	void Replace(const int Index, T& InType)
```

Comment:
This function is a bit too simple, I think. It is unnecessary because you are literally doing DynamicArray[Index] = InType, which the user could just do themselves.